src/schedule/ll_schedule.c

```diff
@@ -126,6 +134,8 @@ static void schedule_ll_tasks_execute(struct ll_schedule_data *sch)
 
 		tr_dbg(&ll_tr, "task %p %pU being started...", task, task->uid);
 
+		task->state = SOF_TASK_STATE_RUNNING;
```

Comment:
Redundant line?

Reply:
This is new added to avoid a pending task to be scheduled multiple times (line 79 checks and filter it out).

Reply:
In principle I don't think it's bad to set the task state to RUNNING for the duration of its handler, but I don't see how not doing it could cause it to be scheduled multiple times. Line 79 is in `schedule_ll_is_pending()` and it is never called concurrently with this function. And when this function isn't running, last time it has returned task's status was overwritten on line 139 below. So in fact I don't think this affects the run-time.

Reply:
Agree with you, as applied below, this is only to make the logic clearer, not to fix any existed/observed issue.